packages/block-editor/src/components/iframe/index.js

```diff
@@ -279,27 +288,22 @@ function Iframe( {
 				srcDoc={ srcDoc }
 				title={ __( 'Editor canvas' ) }
 			>
+				{ head && createPortal( headChildren, head ) }
 				{ iframeDocument &&
 					createPortal(
-						<>
-							<head ref={ headRef }>
-								{ styleAssets }
-								{ head }
-							</head>
-							<body
-								ref={ bodyRef }
-								className={ classnames(
-									'block-editor-iframe__body',
-									'editor-styles-wrapper',
-									...bodyClasses
-								) }
-							>
-								{ contentResizeListener }
-								<StyleProvider document={ iframeDocument }>
-									{ children }
-								</StyleProvider>
-							</body>
-						</>,
+						<body
+							ref={ bodyRef }
+							className={ classnames(
+								'block-editor-iframe__body',
+								'editor-styles-wrapper',
+								...bodyClasses
+							) }
+						>
+							{ contentResizeListener }
+							<StyleProvider document={ iframeDocument }>
+								{ children }
+							</StyleProvider>
+						</body>,
 						iframeDocument.documentElement
```

Comment:
I think it would be good to use the same approach on the `body`:
- Avoid removing the existing iframe body
- Portal to the `body` instead of the `documentElement`.

With the changes in this PR, `documentElement` already contains a `head` child, and the portal is appending a `body`, which is something that I didn't know would work. There's no reference in the react docs to what happens if you create a portal on an element that already has children.

Reply:
Is there any way we could make react use the existing head and body?

Reply:
It does already use the existing head in this PR, but I haven't made it use the existing body. I'd be happy to do that.

Reply:
@ellatrix I've pushed a commit ([7d3a6cb](https://github.com/WordPress/gutenberg/pull/47808/commits/7d3a6cbc69b60d2bdcd2a7b01aa88a03e9753f91)) that keeps the existing `<body>`. The main issue is that there were a number of refs being bound to the body, and now that its no longer rendered as a react element, that's harder to do.

The quick solution I came up with is to add an extra div around the content and bind the refs to that div. I know you hate divs 😄

Reply:
Ah, I see. I have no idea how it works either, but that's an interesting idea. I'll do some research.

Reply:
I tested it out, but I'm not sure it will be possible. `hydrateRoot` really seems like the equivalent of `createRoot` and is expected to be called once to create a completely isolated react tree (what I guess you could consider to be a react application).

So while I could vaguely get the iframe to hydrate, I think it creates a separate react tree in the iframe that is disconnected from the rest of the editor.

I guess this is the main advantage of `createPortal` it ensures the iframe's children are still part of the editor's react tree and so things like using context work correctly.

There's a branch here if you'd like to experiment - https://github.com/WordPress/gutenberg/compare/try/hydrating-the-iframe?expand=1.